torch/distributed/_tensor/placement_types.py

```diff
@@ -54,37 +54,102 @@ def _split_tensor(
             tensor.size(self.dim) >= num_chunks
         ), f"Tensors to be sharded on dim {self.dim} must be at least as large as "
         f"the number of devices in that dimension {num_chunks}"
-        # split tensor over dimension `dim` into n slices with padding if necessary
-        tensor_list = list(tensor.tensor_split(num_chunks, self.dim))
-        idx_start_to_pad = tensor.size(self.dim) % num_chunks
-        if with_padding or contiguous:
+
+        # chunk tensor over dimension `dim` into n slices with padding if necessary
+        tensor_list = list(torch.chunk(tensor, num_chunks, dim=self.dim))
+        # compute the chunk size inline with ``torch.chunk``
+        full_chunk_size = (tensor.size(self.dim) + num_chunks - 1) // num_chunks
+
+        # Compute chunk size for each chunk for ``self.dim``
+        chunk_sizes = [
+            tensor_list[idx].size(self.dim) if idx < len(tensor_list) else 0
+            for idx in range(num_chunks)
+        ]
+        # Get idx start to pad
+        idx_start_to_pad = next(
```

Comment:
do we really need this given that we only need `pad_sizes`? `pad_sizes` can actually be computed without this iiuc.